explore/src/main/scala/explore/tabs/ConstraintsTile.scala

```diff
@@ -18,39 +23,77 @@ import explore.components.ui.ExploreStyles
 import explore.constraints.ConstraintsPanel
 import explore.implicits._
 import explore.model._
-import explore.model.reusability._
 import explore.schemas.ObservationDB
 import japgolly.scalajs.react._
 import japgolly.scalajs.react.vdom.html_<^._
 import lucuma.core.model.ConstraintSet
+import lucuma.core.model.Observation
 import lucuma.ui.reusability._
 import react.common._
 import react.common.implicits._
 import react.semanticui.addons.select.Select
 import react.semanticui.addons.select.Select.SelectItem
+import react.semanticui.elements.button.Button
 import react.semanticui.modules.dropdown.Dropdown
+import react.semanticui.sizes
+
+import scala.util.Random
 
 object ConstraintsTile {
+
+  private def newConstraintSet(
+    obsId:           Observation.Id
+  )(implicit client: TransactionalClient[IO, ObservationDB]): IO[Unit] =
+    for {
+      id <-
+        IO(Random.nextInt(0xfff)).map(int =>
+          ConstraintSet.Id(PosLong.unsafeFrom(int.abs.toLong + 1))
+        )
+      _  <-
+        AddConstraintSet
+          .execute(
+            defaultCreateConstraintSet(
+              defaultConstraintSetResult(id, NonEmptyString.unsafeFrom(s"New Constraints [$id]"))
+            )
+          )
+          .void
+      _  <- AssignConstraintSetToObs.execute(id, obsId)
+    } yield ()
+
   def constraintsTile(
     constraintSetId: Option[ConstraintSet.Id],
     obsSummaryOpt:   Option[ObsSummary],
     constraintsInfo: ConstraintsInfo
   )(implicit ctx:    AppContextIO): Tile = {
     def constraintsSelectorFn(
       constraintSetId: Option[ConstraintSet.Id],
-      obsSummaryOpt:   Option[ObsSummary],
+      obsId:           Observation.Id,
       observations:    ConstraintsInfo
     ): VdomNode =
-      Select(
-        value = constraintSetId.map(_.show).orEmpty, // Set to empty string to clear
-        placeholder = "Select a constraint set",
-        onChange = (a: Dropdown.DropdownProps) =>
-          (obsSummaryOpt, ConstraintSet.Id.parse(a.value.toString)).mapN { (obsId, csId) =>
-            AssignConstraintSetToObs
-              .execute(csId, obsId.id)
-              .runAsyncAndForgetCB *> Callback.log(s"Set to $csId")
-          }.getOrEmpty,
-        options = observations.map(s => new SelectItem(value = s.id.show, text = s.name.value))
+      <.span(
+        Select(
+          value = constraintSetId.map(_.show).orEmpty, // Set to empty string to clear
+          placeholder = "Select a constraint set",
+          // size = sizes.Small,
```

Comment:
Commented code?

Reply:
Ah, good catch. The commented code doesn't work; I left it commented out while investigating adding it to the facade, but React SUI doesn't really support it.